readthedocs/projects/tasks.py

```diff
@@ -717,7 +717,7 @@ def setup_vcs(self, environment):
             # Re raise the exception to stop the build at this point
             raise
 
-        commit = self.commit or self.project.vcs_repo(self.version.slug).commit
+        commit = self.commit or self.get_vcs_repo(environment).commit
```

Comment:
Another approach is to create a local build environment with `record=False` but feels overkill instead of reusing the build env we already have.